WordPress/Classes/Networking/BlogServiceRemoteREST.m

```diff
@@ -236,32 +259,93 @@ - (NSDictionary *)mapPostFormatsFromResponse:(id)response
 
 - (RemoteBlogSettings *)remoteBlogSettingFromJSONDictionary:(NSDictionary *)json
 {
-    NSDictionary *rawSettings = [json dictionaryForKey:BlogRemoteSettingsKey];
+    NSAssert([json isKindOfClass:[NSDictionary class]], @"Invalid Settings Kind");
     
-    RemoteBlogSettings *remoteSettings = [RemoteBlogSettings new];
+    RemoteBlogSettings *settings = [RemoteBlogSettings new];
+    NSDictionary *rawSettings = [json dictionaryForKey:RemoteBlogSettingsKey];
     
-    remoteSettings.name = [json stringForKey:BlogRemoteNameKey];
-    remoteSettings.desc = [json stringForKey:BlogRemoteDescriptionKey];
-    remoteSettings.defaultCategory = [rawSettings numberForKey:BlogRemoteDefaultCategoryKey] ?: @(BlogRemoteUncategorizedCategory);
+    // General
+    settings.name = [json stringForKey:RemoteBlogNameKey];
+    settings.tagline = [json stringForKey:RemoteBlogTaglineKey];
+    settings.privacy = [rawSettings numberForKey:RemoteBlogPrivacyKey];
```

Comment:
I've noticed we're using safe expectations to parse backend responses in few other spots (such as [this one](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Networking/CommentServiceRemoteREST.m#L399) and [this other one](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Networking/MediaServiceRemoteREST.m#L204), among several).

How about if we add assertions to Safe Expectations (so that Beta builds catch the error), still, AppStore builds won't crash, if anything goes wrong?